src/Sulu/Bundle/WebsiteBundle/Controller/ExceptionController.php

```diff
@@ -67,36 +88,45 @@ public function showAction(
             }
         }
 
-        $baseTemplate = $this->findTemplate($request, $request->getRequestFormat(), $code, $showException);
-
+        $showException = $request->attributes->get('showException', $this->debug);
         if ($showException || $request->getRequestFormat() !== 'html' || $template === null) {
-            return new Response($this->twig->render(
-                (string) $baseTemplate,
-                [
-                    'status_code' => $code,
-                    'status_text' => isset(Response::$statusTexts[$code]) ? Response::$statusTexts[$code] : '',
-                    'exception' => $exception,
-                    'logger' => $logger,
-                    'currentContent' => $currentContent,
-                ]
-            ));
+            return $this->exceptionController->showAction($request, $exception, $logger);
         }
 
-        $parameter = [
-            'status_code' => $code,
-            'status_text' => isset(Response::$statusTexts[$code]) ? Response::$statusTexts[$code] : '',
-            'exception' => $exception,
-            'currentContent' => $currentContent,
-        ];
-
-        $data = $this->parameterResolver->resolve($parameter, $this->requestAnalyzer);
+        $context = $this->parameterResolver->resolve(
+            [
+                'status_code' => $code,
+                'status_text' => isset(Response::$statusTexts[$code]) ? Response::$statusTexts[$code] : '',
+                'exception' => $exception,
+                'currentContent' => $this->getAndCleanOutputBuffering($request->headers->get('X-Php-Ob-Level', -1)),
+            ],
+            $this->requestAnalyzer
+        );
 
         return new Response(
             $this->twig->render(
                 $template,
-                $data
+                $context
             ),
             $code
         );
     }
+
+    /**
+     * Returns and cleans output-buffer.
+     *
+     * @param int $startObLevel
+     *
+     * @return string
+     */
+    protected function getAndCleanOutputBuffering($startObLevel)
```

Comment:
That's one of the reasons I didn't use the decorator pattern here... Reimplementing existing stuff is something to avoid.